omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/EncounterProviderResource1_9.java

```diff
@@ -0,0 +1,146 @@
+import org.openmrs.module.webservices.rest.web.api.RestHelperService;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingSubResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@link org.openmrs.module.webservices.rest.web.annotation.Resource} for EncounterProvider, supporting standard CRUD operations
+ */
+@SubResource(path = "encounterprovider", parent = EncounterResource1_9.class, supportedClass = EncounterProvider.class, supportedOpenmrsVersions = {"1.9.*", "1.10.*", "1.11.*", "1.12.*"})
+public class EncounterProviderResource1_9 extends DelegatingSubResource<EncounterProvider, Encounter, EncounterResource1_9> {
+
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+        if (rep instanceof DefaultRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("provider", Representation.REF);
+            description.addProperty("encounterRole", Representation.REF);
+            description.addProperty("voided");
+            description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+            return description;
+        }
+        if (rep instanceof FullRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("provider", Representation.DEFAULT);
+            description.addProperty("encounterRole", Representation.DEFAULT);
+            description.addProperty("voided");
+            description.addProperty("auditInfo", findMethod("getAuditInfo"));
+            description.addSelfLink();
+            return description;
+        }
+        return null;
+    }
+
+    @Override
+    public DelegatingResourceDescription getCreatableProperties() {
+        DelegatingResourceDescription description = new DelegatingResourceDescription();
+        description.addProperty("provider");
+        description.addProperty("encounter");
+        description.addProperty("encounterRole");
+        return description;
+    }
+
+    @Override
+    public DelegatingResourceDescription getUpdatableProperties() {
```

Comment:
You actually don't need this if all creatable fields are also updatable because under the hood getUpdatableProperties calls getCreatableProperties to determine updatable fields